shedlock-provider-jedis/src/main/java/net/javacrumbs/shedlock/provider/jedis/JedisLockProvider.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2009-2017 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.javacrumbs.shedlock.provider.jedis;
+
+import net.javacrumbs.shedlock.core.LockConfiguration;
+import net.javacrumbs.shedlock.core.LockProvider;
+import net.javacrumbs.shedlock.core.SimpleLock;
+import net.javacrumbs.shedlock.support.LockException;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.Instant;
+import java.util.Optional;
+
+/**
+ * Uses Redis's `SET resource-name anystring NX PX max-lock-ms-time` as locking mechanism.
+ *
+ * See https://redis.io/commands/set
+ *
+ */
+public class JedisLockProvider implements LockProvider {
+
+    private static final String keyPrefix = "job-lock";
+
+    private static final String DEFAULT_ENV = "default";
+
+    private JedisPool jedisPool;
+    private String environment;
+
+    public JedisLockProvider(JedisPool jedisPool) {
+        this(jedisPool, DEFAULT_ENV);
+    }
+
+    public JedisLockProvider(JedisPool jedisPool, String environment) {
+        this.jedisPool = jedisPool;
+        this.environment = environment;
+    }
+
+    @Override
+    public Optional<SimpleLock> lock(LockConfiguration lockConfiguration) {
+        long difference = getDifference(lockConfiguration);
+        if (difference > 0) {
+            String key = buildKey(lockConfiguration.getName(), this.environment);
+            try (Jedis jedis = jedisPool.getResource()) {
+                String rez = jedis.set(key, buildValue(), "NX", "PX", difference);
+                if (rez != null && "OK".equals(rez)) {
+                    return Optional.of(new RedisLock(key, jedisPool));
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    long getDifference(LockConfiguration lockConfiguration) {
```

Comment:
Could you please write some unit tests for this method? I suspect it might get simplified by using Java 8 date features. It's a bit hard to read right now.